src/test/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImplTest.java

```diff
@@ -42,53 +36,59 @@ private BitbucketPullRequest setupPR(String newKey) {
 
     @Test
     public void testAddPRWithNewKey() {
-        BitbucketPullRequest bitbucketPullRequest = setupPR(key);
+        BitbucketPullRequest bitbucketPullRequest = setupPR(key, BitbucketPullState.OPEN, 1);
         PullRequestStoreImpl.CacheKey cacheKey = new PullRequestStoreImpl.CacheKey(key, slug, serverId);
 
         pullRequestStore.addPullRequest(serverId, bitbucketPullRequest);
 
-        assertEquals("Size mismatch for map;", pullRequestStore.getPullRequests().size(), 1);
-        assertTrue(pullRequestStore.getPullRequests().containsKey(cacheKey));
-        ConcurrentLinkedQueue queue = (ConcurrentLinkedQueue) pullRequestStore.getPullRequests().get(cacheKey);
-        assertEquals("Size mismatch for queue;", queue.size(), 1);
-        assertTrue(queue.contains(bitbucketPullRequest));
+        assertTrue(pullRequestStore.getPullRequest(key, slug, serverId, bitbucketPullRequest.getId()) == bitbucketPullRequest);
+
     }
 
     @Test
     public void testAddPRWithExistingKey() {
         testAddPRWithNewKey(); //there is an entry in pullRequestStore with same cacheKey
```

Comment:
we tend to not call tests from other tests. It is better that you extract out the relevant bits you want to reuse to a private method and call that instead.
(here and all other places)